### PR TITLE
KCL: Don't track operations for artifact graph within the stdlib.

### PR DESCRIPTION
**Pull request by adamchalmers**
Variables declared within stdlib code shouldn't appear in the feature tree -- users can't read the source code, and those variables are just useless noise to them. For example, using a hole currently adds all this junk to the feature tree:

image

The parameters like `t`, `tr`, `br` etc are all from the internals of the `hole` function and should not be exposed to the user. This PR removes all those internals, showing the user only artifacts from their code.

Screenshot 2025-10-16 at 5 07 19 PM

### CODE REVIEW

File: rust/kcl-lib/src/execution/fn_call.rs [hunk 1]
```diff
@@ -311,7 +323,7 @@ impl FunctionSource {
             // return value. The call takes ownership of the args,
             // so we need to build the op before the call.
             exec_state.push_op(op);
-        } else if !self.is_std {
+        } else if !is_calling_into_stdlib && !self.is_std {
```
Comment (jtran) [inline rust/kcl-lib/src/execution/fn_call.rs]:
```suggestion
        } else if !is_calling_into_stdlib {
```

File: rust/kcl-lib/tests/array_concat_non_array/ops.snap [hunk 2]
```diff
@@ -4,126 +4,12 @@ description: Operations executed array_concat_non_array.kcl
...
-          }
-        }
-      },
-      "nodePath": {
-        "steps": []
-      },
-      "sourceRange": []
-    },
     {
       "type": "GroupEnd"
```
Comment (jtran) [inline rust/kcl-lib/tests/array_concat_non_array/ops.snap:46]: I wouldn't have expected the GroupEnd to still be here.

File: rust/kcl-lib/src/execution/exec_ast.rs [hunk 3]
```diff
@@ -2669,6 +2669,13 @@ y = x[0mm + 1]
         parse_execute(&ast).await.unwrap();
     }
 
+    #[cfg(not(feature = "artifact-graph"))]
+    #[tokio::test(flavor = "multi_thread")]
+    async fn no_artifacts_from_within_hole_call() {
+        panic!("This test must be run with the 'artifact-graph' feature.")
```
Comment (jtran) [inline rust/kcl-lib/src/execution/exec_ast.rs]: Doesn't this mean that you can no longer run `cargo test` because it will always fail? I think that in the past, we just didn't run the test if the feature wasn't enabled. See `rust/kcl-lib/e2e/executor/cache.rs`.

Comment (adamchalmers) [reply rust/kcl-lib/src/execution/exec_ast.rs]: Yeah that's reasonable. Done.

File: rust/kcl-lib/tests/array_concat_non_array/ops.snap (hunk 2 above)
Comment (adamchalmers) [reply rust/kcl-lib/tests/array_concat_non_array/ops.snap:46]: Good point, I think I've resolved it, see latest commit a1532c6974e1f4b398b59e778789d432951c54b1

File: rust/kcl-lib/tests/kcl_samples/bone-plate/ops.snap [hunk 4]
```diff
@@ -171,56 +171,6 @@ description: Operations executed bone-plate.kcl
       },
       "sourceRange": []
     },
-    {
```
Comment (jtran) [inline rust/kcl-lib/tests/kcl_samples/bone-plate/ops.snap:174]: This is triggering the edge case I mentioned. This is a call to a user-defined function. But it's happening inside `map()`, i.e. calling from inside stdlib. My intent was to include these.

Comment (adamchalmers) [reply rust/kcl-lib/tests/kcl_samples/bone-plate/ops.snap:174]: OK... I'm gonna have to come back to this tomorrow, too tired now :)

Comment (adamchalmers) [reply rust/kcl-lib/tests/kcl_samples/bone-plate/ops.snap:174]: I think I've fixed it, please lmk
